### DISCUSSION TITLE
How to control floor heating actuator channels?

### DISCUSSION THREAD

**Discussion by nicolazilio** [OpenCCU/OpenCCU#2398] [🇺🇸 Q&A] [answered]
Dear all,

I have a HmIP floor heating actuator (HmIP-FAL230-C6) that I’ve added to HA via the RaspberryMatic add-on and the Homematic(IP) Local integration.

Is there any way to control the channels/valves directly from HA? I (believe) can see them exposed as binary sensor, e.g. binary_sensor.floor_heating_actuator_ch1, but when I toggle the state of any of those channels from off to on, the corresponding valve does not open. Is control of these valves only possible from the RaspberryMatic add-on?

Thanks in advance!

**Comment by jens-maus** [comment, marked as answer]
No there is unfortunately NO way to directly control the homematicIP floor heating actuators (close/open a valve, etc.) like the one you reference in your question. All these floor heating actors are meant to be only linked to the corresponding homematicIP wall temperature devices as well as to other HmIP driven actors which e.g. start/stop your floor heating pump in case it identifies that this is necessary.

long story short: there is indeed NO way to directly control a HmIP floor heating actor, not via HomeAssistant, not via RaspberryMatic ans also not via a standard CCU3. A RaspberryMatic central can also "just" control a linked wall thermostat by raising the target temperature for the room so that the floor heating actor will then decide to open/close the valve by itself.

So get a HmIP wall thermostat sensor, link it to your floor heating actor and then you can change the taget temperature entity of that wall thermostat sensor and it will then directly communicate to the floor heating actor which then does the rest.

**Reply by nicolazilio** [reply]
Thanks for the clarification. I have three HmIP temperature sensors/thermostats, so that is no problem.

Just to double check, what I should do is the following:

1. Create a direct device connection in the Raspberrymatic Add-on (Home page > Programs and connections > Direct device connections) where the transmitter is the thermostat/temperature sensor, specifically the `Connection partner floor heating` for this entity, which acts as the transmitter and a specific floor heating actuator channel as the receiver. I am guessing, I can leave the default settings for this connection as they are for now, for example

![image](https://github.com/jens-maus/RaspberryMatic/assets/2800360/573cbe5c-a6dd-494e-914a-4555672d55b0)

2. From Home Assistant itself, I can set the corresponding `climate` entity to a desired temperature, if this is higher than the actual measured temperature in the room, then the system automatically opens the valve connected to that `climate` entity. Correct?

Thanks!

**Reply by jens-maus** [reply]
> 1. Create a direct device connection in the Raspberrymatic Add-on (Home page > Programs and connections > Direct device connections) where the transmitter is the thermostat/temperature sensor, specifically the `Connection partner floor heating` for this entity, which acts as the transmitter and a specific floor heating actuator channel as the receiver. I am guessing, I can leave the default settings for this connection as they are for now, for example

This is more or less correct, yes. But in the RaspberryMatic/CCU world you can make your life way easier if you use the heating group functionality in the RaspberryMatic/CCU WebUI. That means go to Settings -> Groups and create heating groups (e.g. create one group per room). And then within these room-based heating groups you can add the different heating actuators and sensors and RaspberryMatic/CCU will then automatically take care to create all these direct linkage between all HmIP devices. E.g. you can also add window/door sensors to these groups as well and then upon an open window the wall thermostat will automatically switch to a special "window open temperature" to put down the room target temperature, thus close the valves if necessary.

> 2. From Home Assistant itself, I can set the corresponding `climate` entity to a desired temperature, if this is higher than the actual measured temperature in the room, then the system automatically opens the valve connected to that `climate` entity. Correct?

That's correct! But keep in mind that it won't immediately close the valves beause all these HmIP heating actuator have own internal logic how, when and how much they will close/open their valves to reach the target temperature. Most of this internal behaviour you can adjust by adjusting the individual device settings and also the settings of the direct link connections, but there is some additional logic (especially in the floor heating actuators) that might come on top.

**Reply by nicolazilio** [reply]
To summarize, your recommendation is to create a heating group and add the desired heating schedule to it all within RaspberryMatic. However, should one wish to, one can also control the state of these RaspberryMatic heating groups from HA using the climate entity attached to the relevant temperature sensor; for example, if one specific evening I want to turn off the heating.

Thanks!
